src/types/contracts/settingsController.ts

```diff
@@ -0,0 +1,34 @@
+export type Delegate = {
+  address: string; // wallet address like juno1dg2...
+  expires?: number; // datetime int of delegation expiry
+};
+
+export type SettingsPermissions = {
+  owner_controlled: boolean;
+  gov_controlled: boolean;
+  modifiable: boolean;
+  delegate?: Delegate;
+};
+
+export type EndowmentController = {
+  endowment_controller: SettingsPermissions;
+  name: SettingsPermissions;
+  image: SettingsPermissions;
+  logo: SettingsPermissions;
+  categories: SettingsPermissions;
+  kyc_donors_only: SettingsPermissions;
+  split_to_liquid: SettingsPermissions;
+  ignore_user_splits: SettingsPermissions;
+  donation_match_active: SettingsPermissions;
+  beneficiaries_allowlist: SettingsPermissions;
+  contributors_allowlist: SettingsPermissions;
+  maturity_allowlist: SettingsPermissions;
+  earnings_fee: SettingsPermissions;
+  deposit_fee: SettingsPermissions;
+  withdraw_fee: SettingsPermissions;
+  aum_fee: SettingsPermissions;
+};
+
+export type UpdateEndowmentControllerMsg = Partial<EndowmentController> & {
```

Comment:
though `type EndowmentController` and `type UpdateEndowmentControllerMsg` looks the same, they are actually separate structs in the contracts. so it's better to define them separately here.  Moreover, they do have some differences on them 
e.g `type EndowmentController` have `profile` field while `UpdateEndowmentControllerMsg` have `donation_match_active` field

Reply:
@ap-justin @misicnenad This is changing as we speak! I've been working on harmonizing that and some other bug fixes this morning. Stand by for an update later this afternoon. :+1:

Reply:
**UPDATE:**
`profile` is removed now from the controller and update msg struct & `donation_match_active` has been added to both. The "profile related" fields (ie. `kyc_donors_only, name, image, logo, categories`) have had the default values changed for Charity type endowments so that they ARE modifiable should they choose to do so.

Reply:
@ap-justin 
> though type EndowmentController and type UpdateEndowmentControllerMsg may look the same, they are actually separate structs in the contracts

In light of recent changes to types, we can keep these types as-is until they start drastically differing from each other.

Reply:
> The "profile related" fields (ie. kyc_donors_only, name, image, logo, categories) have had the default values changed for Charity type endowments so that they ARE modifiable should they choose to do so

@SovereignAndrey does that mean we **should** have the "Actions" (lock(ed) forever btn) column displayed for charities as well (previous requirement was to hide it for them https://github.com/AngelProtocolFinance/angelprotocol-web-app/issues/1708#issuecomment-1463675056)?

Reply:
> > The "profile related" fields (ie. kyc_donors_only, name, image, logo, categories) have had the default values changed for Charity type endowments so that they ARE modifiable should they choose to do so
> 
> @SovereignAndrey does that mean we **should** have the "Actions" (lock(ed) forever btn) column displayed for charities as well (previous requirement was to hide it for them [#1708 (comment)](https://github.com/AngelProtocolFinance/angelprotocol-web-app/issues/1708#issuecomment-1463675056))?

Yes, we should show it after all, since not everything it locked forever.